File: ProjetoJava/src/interfaces.java
interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    public void imprimir() {
        System.out.println("Imprimindo relatório...");
    }
}

class Contrato implements Imprimivel {
    public void imprimir() {
        System.out.println("Imprimindo contrato...");
    }

    public static void main(String[] args) {
        Imprimivel[] documentos = { new Relatorio(), new Contrato() };
        for (Imprimivel doc : documentos) {
            doc.imprimir();
        }
    }
}

File: GoLangProject/interfaces.go
package main

import "fmt"

type Imprimivel interface {
	Imprimir()
}

type Relatorio struct{}

func (r Relatorio) Imprimir() {
	fmt.Println("Imprimindo relatório...")
}

type Contrato struct{}

func (c Contrato) Imprimir() {
	fmt.Println("Imprimindo contrato...")
}

func main() {
	documentos := []Imprimivel{Relatorio{}, Contrato{}}
	for _, doc := range documentos {
		doc.Imprimir()
	}
}
